ansible_collections/arista/avd/plugins/action/anta_runner.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import json
+
+from ansible.errors import AnsibleActionFail
+from ansible.plugins.action import ActionBase
+from anta.cli.utils import setup_logging
+from anta.inventory.models import InventoryDeviceHttpApi
+from anta.loader import parse_catalog
+from anta.result_manager import ResultManager
+from yaml import safe_load
+
+from ansible_collections.arista.avd.plugins.plugin_utils.eos_validate_state_utils import update_catalog
+
+
+class ActionModule(ActionBase):
+    def run(self, tmp=None, task_vars=None):
+        if task_vars is None:
+            task_vars = {}
+
+        result = super().run(tmp, task_vars)
+        del tmp  # tmp no longer has any effect
+
+        play_context = self._play_context
+        connection = self._connection.httpapi
+
+        # Need to set the right become_context to the _connection.httpapi object
+        connection.set_become(play_context)
+
+        try:
+            # Setup ANTA logging
+            anta_logging = task_vars.get("anta_logging")
+            if anta_logging:
+                setup_logging(level=anta_logging)
+
+            # Creating the ANTA device object with the HttpApi interface
+            anta_device = InventoryDeviceHttpApi(session=connection)
+
+            # Creating the catalog from the Ansible task
+            catalog_path = self._task.args.get("catalog_path")
+            with open(catalog_path, "r", encoding="UTF-8") as file:
+                input_catalog = safe_load(file)
+
+            # Raises error if the provided ANTA test catalog is empty
+            if not input_catalog:
+                raise AnsibleActionFail(f"No tests were provided in the catalog: {catalog_path}")
+
+            # In ANTA, tests requiring parameters are "skipped" if they are not provided or not found in task_vars
```

Comment:
I would prefer not to rely on this behavior, but instead raise an error if a test is set in the input_catalog, but we cannot find the required variables for it.